### ISSUE TITLE
Calico Helm chart doesn't support nodeAffinity or nodeSelector configuration

### ISSUE THREAD

**Issue by szpuni** [aws/eks-charts#177]
Currently calico helm chart do not allow for nodeSelector or nodeAffinity manipulation. Tolerations are only available in one of deployments are are hardcoded same way as 2 previous attributes.

Above makes impossible to choose nodes in cluster on which calico is installed.

Deployment.yaml file needs to be updated to allow using custom values in nodeSelector, nodeAffinity, tollerations:

```
    spec:
      priorityClassName: system-cluster-critical
      nodeSelector:
        beta.kubernetes.io/os: linux
      tolerations:
        # Mark the pod as a critical add-on for rescheduling.
        - key: CriticalAddonsOnly
          operator: Exists
```

**Comment by limed** [comment]
Resolved in #260

**Comment by haouc** [comment]
@szpuni if you still have concerns, please be aware of our latest announcement. Since we are deprecating Calico charts in this repo, we recommend you to check our [important announcement](https://github.com/aws/eks-charts/tree/master/stable/aws-calico) and suggested [installation instruction](https://docs.aws.amazon.com/eks/latest/userguide/calico.html) for required or latest Calico version. You can also refer to [Calico configuration reference](https://projectcalico.docs.tigera.io/reference/installation/api).
If you need migrate Calico from non-operator based installation to operator based installation, we recommend you to check [this Tigera document](https://projectcalico.docs.tigera.io/maintenance/operator-migration). 
If you need upgrade you calico operator but you didn't install Calico via helm, you can check [this Tigera document](https://projectcalico.docs.tigera.io/getting-started/kubernetes/quickstart). Please being aware that any customized installation may encounter compatibility issue during upgrading.
If you have any issues or questions during Calico resources installation/upgrading/deletion, please refer to the [README](https://github.com/aws/eks-charts/tree/master/stable/aws-calico) and send request to Calico repositories.
I am closing this ticket due to unsupported charts upgrade. Please feel free to send questions to our triage email ([k8s-awscni-triage@amazon.com](mailto:k8s-awscni-triage@amazon.com)).
